app/products/calls/components/audio_device_button.tsx

```diff
@@ -0,0 +1,141 @@
+            setPreferredAudioRoute(device);
+            dismissBottomSheet();
+        };
+
+        const renderContent = () => {
+            return (
+                <View>
+                    {available.includes(AudioDevice.Earpiece) &&
+                        <SlideUpPanelItem
+                            icon={'phone-in-talk'}
+                            onPress={() => selectDevice(AudioDevice.Earpiece)}
+                            text={earpieceLabel}
+                            textStyles={currentDevice === AudioDevice.Earpiece ? {...bold, color} : {}}
+                        />
+                    }
+                    {available.includes(AudioDevice.Speakerphone) &&
+                        <SlideUpPanelItem
+                            icon={'volume-high'}
+                            onPress={() => selectDevice(AudioDevice.Speakerphone)}
+                            text={speakerLabel}
+                            textStyles={currentDevice === AudioDevice.Speakerphone ? {...bold, color} : {}}
+                        />
+                    }
+                    {available.includes(AudioDevice.Bluetooth) &&
+                        <SlideUpPanelItem
+                            icon={'cellphone'}
+                            onPress={() => selectDevice(AudioDevice.Bluetooth)}
+                            text={bluetoothLabel}
+                            textStyles={currentDevice === AudioDevice.Bluetooth ? {...bold, color} : {}}
+                        />
+                    }
+                </View>
+            );
+        };
+
+        await bottomSheet({
+            closeButtonId: 'close-other-actions',
+            renderContent,
+            snapPoints: [1, bottomSheetSnapPoint(audioDeviceInfo.availableAudioDeviceList.length + 1, ITEM_HEIGHT, bottom)],
+            title: intl.formatMessage({id: 'mobile.calls_output_options', defaultMessage: 'Choose preferred output'}),
+            theme,
+        });
+    }, [setPreferredAudioRoute, audioDeviceInfo, bold, color]);
+
+    const toggleSpeakerPhone = useCallback(() => {
+        setSpeakerphoneOn(!currentCall?.speakerphoneOn);
+    }, [currentCall?.speakerphoneOn]);
+
+    if (Platform.OS === 'ios') {
+        return (
+            <Pressable
+                testID={'toggle-speakerphone'}
+                style={[style.button, isLandscape && style.buttonLandscape]}
+                onPress={toggleSpeakerPhone}
+            >
+                <CompassIcon
+                    name={'volume-high'}
+                    size={24}
+                    style={[
+                        style.buttonIcon,
+                        isLandscape && style.buttonIconLandscape,
+                        style.speakerphoneIcon,
+                        currentCall.speakerphoneOn && style.buttonOn,
+                    ]}
+                />
+                <Text style={style.buttonText}>{speakerLabel}</Text>
+            </Pressable>
+        );
```

Comment:
it looks to me that this can be another component and you pass the correct props based on the Platform

Reply:
Good idea, separated them into platform files, much cleaner now.